### PR TITLE
add Dockerfile for C#

### PR DESCRIPTION
**Pull request by jtattermusch**
Installs:
-- mono
-- libgrpc from jessie backports 
-- C# native extension from deb package downloaded from github

Tested with greeter and routeguide.

### CODE REVIEW

**Comment by tbetbetbe** [comment]
Sweet!!

File: 0.11/csharp/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+RUN echo "deb http://download.mono-project.com/repo/debian wheezy-libtiff-compat main" | tee -a /etc/apt/sources.list.d/mono-xamarin.list
+RUN apt-get update && apt-get install -y mono-complete mono-devel nuget && apt-get clean
+
+# Install trusted root certificates needed to restore NuGet packages
+RUN mozroots --import --sync
+
+RUN apt-get update && apt-get install -y -q wget libgrpc0 && apt-get clean
+
+# Download and install C# native extension
+RUN cd /root && wget https://github.com/grpc/grpc/releases/download/release-0_11_0/libgrpc-csharp-ext0_0.11.0.0-1_amd64.deb \
```
Comment (tbetbetbe) [inline 0.11/csharp/Dockerfile]: can you use a couple of env vars for the 
- the release branch (release_0_11_0)
- the package version (0.11.0.0-1_amd64)

?
that reduces the risk of typos in future updates

File: 0.11/csharp/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+RUN echo "deb http://download.mono-project.com/repo/debian wheezy main" | tee /etc/apt/sources.list.d/mono-xamarin.list
+RUN echo "deb http://download.mono-project.com/repo/debian wheezy-apache24-compat main" | tee -a /etc/apt/sources.list.d/mono-xamarin.list
+RUN echo "deb http://download.mono-project.com/repo/debian wheezy-libjpeg62-compat main" | tee -a /etc/apt/sources.list.d/mono-xamarin.list
+RUN echo "deb http://download.mono-project.com/repo/debian wheezy-libtiff-compat main" | tee -a /etc/apt/sources.list.d/mono-xamarin.list
+RUN apt-get update && apt-get install -y mono-complete mono-devel nuget && apt-get clean
+
+# Install trusted root certificates needed to restore NuGet packages
+RUN mozroots --import --sync
+
+RUN apt-get update && apt-get install -y -q wget libgrpc0 && apt-get clean
```
Comment (tbetbetbe) [inline 0.11/csharp/Dockerfile]: Here, can you install a [specific version](https://github.com/grpc/grpc-docker-library/blob/master/0.10/python/Dockerfile#L37) of libgrpc0 ?

**Comment by jtattermusch** [comment]
Good comments, thanks. PTAL.

File: 0.11/csharp/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+# Install trusted root certificates needed to restore NuGet packages
+RUN mozroots --import --sync
+
+ENV GRPC_VERSION=0.11.0.0-1
+ENV GRPC_VERSION_BACKPORTS=${GRPC_VERSION}~bpo8+2
+
+RUN apt-get update && apt-get install -y -q wget libgrpc0=${GRPC_VERSION_BACKPORTS} && apt-get clean
+
+# Download and install C# native extension
+RUN cd /root && wget https://github.com/grpc/grpc/releases/download/release-0_11_0/libgrpc-csharp-ext0_${GRPC_VERSION}_amd64.deb \
```
Comment (tbetbetbe) [inline 0.11/csharp/Dockerfile]: one more var for 'release_0_11_0' in the wget uri ?

Comment (jtattermusch) [inline 0.11/csharp/Dockerfile]: Done.

**Comment by jtattermusch** [comment]
PTAL.

**Pull request merged**